src/group_theory/group_algebra.lean

```diff
@@ -0,0 +1,273 @@
+/-
+Copyright (c) 2021 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+
+import algebra.monoid_algebra.basic
+import tactic.basic
+
+/-!
+# Group algebras
+
+This file defines the group algebra `k[G]` of a group `G` with coefficients in a semiring `k`. A
+`G`-module with a compatible `k`-module structure is an `k[G]`-module.
+
+We develop an API allowing us to show `k[Gⁿ]` is a free `k[G]`-module for `n ≥ 1`. This
+will be used to construct a projective resolution of the trivial `k[G]`-module `k`.
+
+## Main definitions
+
+* `group_algebra.has_scalar`: if a group `G` acts on a `k`-module `M` then there is an induced
+action of the group algebra `k[G]` on `M`.
+* `group_algebra.mk_linear`: defines the `k[G]`-linear map induced by a `k`-linear map which is
+also `G`-linear, when the `k` and `G` actions are compatible.
+* `group_algebra.is_scalar_tower'`: the actions of `k` and `k[G]` on `k[Gⁿ]` are compatible.
+
+## Implementation notes
+
+Although `group_algebra k G` is just `monoid_algebra k G`, we make the definition `group_algebra`
+so we can separate material relevant to group cohomology into the namespace `group_algebra`.
+
+We don't want the group cohomology relevant instances defined here to apply to `monoid_algebra`, so
```

Comment:
```suggestion
We don't want the group cohomology relevant instances defined here to apply to `monoid_algebra` as
they cause diamonds with other instances, so
```
It's worth noting that this group cohomology work also has the same diamonds, but we've just decided to put them in this renaming "sandbox" to limit the scope of their effect